modules/consensus/accept.go

```diff
@@ -33,57 +30,29 @@ func (cs *ConsensusSet) validHeader(tx dbTx, b types.Block) error {
 
 	// Check if the block is already known.
 	blockMap := tx.Bucket(BlockMap)
+	if blockMap == nil {
+		return errNoBlockMap
+	}
 	if blockMap.Get(id[:]) != nil {
 		return modules.ErrBlockKnown
 	}
 
 	// Check for the parent.
-	parentBytes := blockMap.Get(b.ParentID[:])
+	parentID := b.ParentID
+	parentBytes := blockMap.Get(parentID[:])
 	if parentBytes == nil {
 		return errOrphan
 	}
+
 	var parent processedBlock
 	err := cs.marshaler.Unmarshal(parentBytes, &parent)
 	if err != nil {
 		return err
 	}
+	// Check that the timestamp is not too far in the past to be acceptable.
+	minTimestamp := cs.blockRuleHelper.minimumValidChildTimestamp(blockMap, &parent)
 
-	// Check that the target of the new block is sufficient.
-	if !checkTarget(b, parent.ChildTarget) {
-		return modules.ErrBlockUnsolved
-	}
-
-	// Check that the timestamp is not too far in the past to be
-	// acceptable.
-	if earliestChildTimestamp(blockMap, &parent) > b.Timestamp {
-		return errEarlyTimestamp
-	}
-
-	// Check that the block is below the size limit.
-	if uint64(len(cs.marshaler.Marshal(b))) > types.BlockSizeLimit {
-		return errLargeBlock
-	}
-
-	// Check if the block is in the extreme future. We make a distinction between
-	// future and extreme future because there is an assumption that by the time
-	// the extreme future arrives, this block will no longer be a part of the
-	// longest fork because it will have been ignored by all of the miners.
-	if b.Timestamp > cs.clock.Now()+types.ExtremeFutureThreshold {
-		return errExtremeFutureTimestamp
-	}
-
-	// Verify that the miner payouts are valid.
-	if !checkMinerPayouts(b, parent.Height+1) {
-		return errBadMinerPayouts
-	}
-
-	// Check if the block is in the near future, but too far to be acceptable.
-	// This is the last check because it's an expensive check, and not worth
-	// performing if the payouts are incorrect.
-	if b.Timestamp > cs.clock.Now()+types.FutureThreshold {
-		return errFutureTimestamp
-	}
-	return nil
+	return cs.blockValidator.ValidateBlock(b, minTimestamp, parent.ChildTarget, parent.Height+1)
```

Comment:
Sorry, I'm still not quite clear. Is the issue that you feel like the refactoring makes sense but the function names need to be revised? Or are you saying that even with clearer function names, the refactoring does not make sense?

The implementation in master is that `validHeader` validates both the block **and** the block header, correct?

Would it make sense to call both functions `validateBlock`? (i.e. `ConsensusSet.validateBlock` and `blockValidator.validateBlock`)?